engine/common/provider/engine.go

```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 
 	"github.com/rs/zerolog"
-	"github.com/vmihailenco/msgpack"
+	"github.com/vmihailenco/msgpack/v4"
```

Comment:
Could we double check the existing version and v4 are encoding into the same bytes data? Do we need some test case to verify?